bundles/org.openhab.core/src/main/java/org/openhab/core/cache/lru/DataRetrievalException.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.cache.lru;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Custom exception class, launched when we cannot retrieve data from the cache entry supplier
+ *
+ * @author Gwendal Roulleau - Initial Contribution
+ *
+ */
+@NonNullByDefault
+public class DataRetrievalException extends RuntimeException {
```

Comment:
I don't like it neither, but I did not find a proper solution.

I made it a runtime exception because, as you can see in the TTSLRUCacheImpl  (line 114), it's the only way I can think of to help handling exception from a Supplier that can launch an exception (such as the TTSService)

I can also completely drop this DataRetrievalException and let the service using the cache cope with its own exception on its own term.

Do you have a better idea ?

Reply:
Eventually, I deleted the DataRetrievalException.
As a result, the code now uses a RuntimeException to wrap the checked TTSException that must be catched inside the supplier get function.
Don't like it either, but it sounds better than a custom RuntimeException with no real use